addons/l10n_in/models/account_invoice.py

```diff
@@ -67,6 +68,16 @@ def _compute_l10n_in_state_id(self):
             else:
                 move.l10n_in_state_id = False
 
+    @api.depends('move_type', 'line_ids.tax_tag_ids', 'state', 'partner_id.l10n_in_pan')
+    def _compute_l10n_in_edi_display_null_pan_warning(self):
+        for move in self:
+            move.l10n_in_display_null_pan_warning = (
+                    move.country_code == 'IN'
+                    and not move.partner_id.l10n_in_pan
+                    and move.state == 'draft'
+                    and move.line_ids.mapped("tax_tag_ids") & (move.env.ref("l10n_in.tax_tag_base_tcs") + move.env.ref("l10n_in.tax_tag_base_tds"))
```

Comment:
`and move.line_ids.mapped("tax_tag_ids") & (move.env.ref("l10n_in.tax_tag_base_tcs") + move.env.ref("l10n_in.tax_tag_base_tds"))`

This line is not so clear. What does the `&` do?

Reply:
Hello @yosuanicolaus ,
The '&' operator returns a set containing only the elements that are common to both sets on its left and right sides. Therefore, it will return the tax tags that are present in both the invoice and the base TDS/TCS tags, if any exist in the invoice and then it return True or False according to it.